include/openthread/thread.h

```diff
@@ -500,6 +500,19 @@ const otIp6Address *otThreadGetLinkLocalAllThreadNodesMulticastAddress(otInstanc
  */
 const otIp6Address *otThreadGetRealmLocalAllThreadNodesMulticastAddress(otInstance *aInstance);
 
+/**
+ * This function retrieves the Service ALOC for given Service ID.
+ *
+ * @param[in]   aInstance    A pointer to an OpenThread instance.
+ * @param[in]   aServiceId   Service ID to get ALOC for.
+ * @param[out]  aServiceAloc A pointer to the Service ALOC.
```

Comment:
We can add in documentation the requirement that `aServiceAloc` must not be null and then skip the check  and error `INVALID_ARGS`. This is a common requirement used in many of the OT APIs (e.g., we never check `aInstance`)  to avoid the overhead of extra null checks.

```suggestion
 * @param[out]  aServiceAloc A pointer to output the Service ALOC. MUST NOT be NULL. 
 ```

Reply:
Changed it in the PR, thanks